pkg/ruler/scheduler.go

```diff
@@ -169,7 +169,7 @@ func (s *scheduler) poll() (map[string]configs.View, error) {
 		return nil, err
 	}
 	s.latestMutex.Lock()
-	s.latestConfig = cfgs.GetLatestConfigID()
+	s.latestConfig = cfgs.GetLatestConfigID(configID)
```

Comment:
I don't like introducing this coupling between caller and callee.
I think it might be simpler to surround the call with `if len(cfgs.Configs) > 0`